src/openrct2/network/TcpSocket.cpp

```diff
@@ -210,11 +210,11 @@ class TcpSocket final : public ITcpSocket
                     (struct sockaddr*)&client_addr, client_len, hostName, sizeof(hostName), nullptr, 0,
                     NI_NUMERICHOST | NI_NUMERICSERV);
                 SetTCPNoDelay(socket, true);
-                tcpSocket = new TcpSocket(socket);
                 if (rc == 0)
                 {
                     _hostName = std::string(hostName);
                 }
+                tcpSocket = new TcpSocket(socket, _hostName);
```

Comment:
I think there is another issue, `_hostName` should not be overwritten for the listener.
I think it should be something along the lines of:
```
                if (rc == 0)
                {
                    tcpSocket = new TcpSocket(socket, hostName);
                }
                else
                {
                    tcpSocket = new TcpSocket(socket);
                }
```